### PR TITLE
improvements in example-msg-generator

### PR DESCRIPTION
**Pull request by nobles**
In this patch I added two changes to `example-msg-generator`. With these changes, `example-msg-generator` can be used for testing purposes easier.

- First message is sent immediately: we do not need to wait `freq`. With this, the tests can speed up.
- With the new `num` option, users can limit how many messages should be generated. Zero means infinite messages (this is the default). Positive integer means the maximum number of messages.

Example:
```
source { example-msg-generator(freq(1) num(5)); };
```

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
do nothing -> CI won't start)

**Comment by furiel** [comment]
@kira-syslogng ok to test

File: modules/examples/sources/msg-generator/msg-generator-source.c [hunk 1]
```diff
@@ -55,14 +56,23 @@ _start_timer(MsgGeneratorSource *self)
...
+
 static gboolean
 _init(LogPipe *s)
 {
   MsgGeneratorSource *self = (MsgGeneratorSource *) s;
   if (!log_source_init(s))
     return FALSE;
-
-  _start_timer(self);
+  if (self->options->max_num)
```
Comment (Kokan) [inline modules/examples/sources/msg-generator/msg-generator-source.c]: Please check the same way as below (` ..max_num < 0`).

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: modules/examples/sources/msg-generator/msg-generator-source.c (hunk 1 above)
Comment (nobles) [reply modules/examples/sources/msg-generator/msg-generator-source.c]: Thanks. Fixed.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**